client/src/components/Profile.tsx

```diff
@@ -67,39 +70,39 @@ export default function Profile () {
     }
   }
 
-  async function handleAllergenSubmit (e) {
+  async function handleAllergenSubmit (e: React.FormEvent<HTMLFormElement>) {
     e.preventDefault();
     try {
-      const newAllergen = allergenRef.current.value;
-      if (newAllergen.length > 1) {
+      const newAllergen = allergenRef.current!.value;
+      if (newAllergen.length > 1 && profile) {
         if (!profile.allergens.includes(newAllergen)) mutation.mutate({
           uid: profile.uid,
           allergens: [...profile.allergens, newAllergen]
         });
-        allergenRef.current.value = '';
+        allergenRef.current!.value = '';
       }
     } catch (error) {
       console.error(error);
     }
   }
 
-  async function handleEventSubmit (e) {
+  async function handleEventSubmit (e: React.FormEvent<HTMLFormElement>) {
     e.preventDefault();
-    const newAllergens = [...new Set([...allergens, ...profile.allergens])];
+    const newAllergens = [...new Set([...allergens, ...profile!.allergens])];
     const menu = await getMenu(newAllergens);
     eventMutation.mutate({
       type,
       allergens: newAllergens,
       members,
-      date: dateRef.current.value,
+      date: dateRef.current!.value,
       menu
     });
     navigate('/');
   }
 
-  function handleMembers (user) {
+  function handleMembers (user: IUser) {
     if (members.length === 0) {
-      members.push(profile.name);
+      profile && members.push(profile.name);
```

Comment:
Would it make more sense to the profile check in the if statement or does that not work here?

Reply:
yep changed